### DISCUSSION TITLE
Tailwind lsp stopped working after last tailwindcss npm update

### DISCUSSION THREAD

**Discussion by bcl1713** [tailwindlabs/tailwindcss#9641] [Help]
Not sure if the problem lies with tailwindlabs/tailwindcss-intellisense or here, but that repo doesn't have a discussion section.  Sometime in the last few days, auto completion has stopped working on react projects.

If I make an empty folder, drop in a tailwind.config.js and an index.html everything works fine.

If I follow the procedures [here](https://tailwindcss.com/docs/guides/create-react-app) to create a brand new react app and add tailwind support, the lsp fails with the following error:
 
`[ERROR][2022-10-23 09:46:26] ...lsp/handlers.lua:485	"Tailwind CSS: Cannot read properties of undefined (reading 'modifier')\nTypeError: Cannot read properties of undefined (reading 'modifier')\n    at api.addVariant (/home/brian/tmp/test-app/node_modules/tailwindcss/lib/lib/setupContextUtils.js:578:40)\n    at /home/brian/tmp/test-app/node_modules/tailwindcss/lib/lib/setupContextUtils.js:538:38\n    at /home/brian/.local/share/nvim/mason/packages/tailwindcss-language-server/node_modules/@tailwindcss/language-server/bin/tailwindcss-language-server:695:4338\n    at Array.forEach (<anonymous>)\n    at QMe (/home/brian/.local/share/nvim/mason/packages/tailwindcss-language-server/node_modules/@tailwindcss/language-server/bin/tailwindcss-language-server:695:3823)\n    at /home/brian/.local/share/nvim/mason/packages/tailwindcss-language-server/node_modules/@tailwindcss/language-server/bin/tailwindcss-language-server:695:283\n    at Generator.next (<anonymous>)\n    at i (/home/brian/.local/share/nvim/mason/packages/tailwindcss-language-server/node_modules/@tailwindcss/language-server/bin/tailwindcss-language-server:2:1059)\n    at processTicksAndRejections (node:internal/process/task_queues:96:5)"`

This is on neovim 0.8.0 on Manjaro. tailwindlabs/tailwindcss version 3.2.1 and tailwindlabs/tailwindcss-intellisense version 0.0.10

Any help would be very much appreciated!

**Comment by bcl1713** [comment]
I can confirm that if I roll the npm package back to version 3.1.8 the error goes away and everything works fine.

**Comment by RobertYim** [comment]
Same here, latest version failed on neovim + lsp, had to roll back to 3.1.8

**Comment by nathanlong** [comment]
@bcl1713 and @RobertYim, in case you haven't gotten an answer (or for fellow google search results for that error message) it looks like they updated the LSP to resolve the issue: https://github.com/tailwindlabs/tailwindcss-intellisense/issues/641

I was experiencing the same issue with nvim 0.8.0 with neovim/nvim-lspconfig, hrsh7th/nvim-cmp, hrsh7th/cmp-nvim-lsp trying to run tailwindcss 3.2.4.

I had the LSP globally installed through npm, but it was at version 0.0.7. Seems like ^0.9.0 will fix it. I updated with `npm install -g @tailwindcss/language-server` and my auto-completion is now working with tailwindcss 3.2.4.
